docs/content/chapter_3.md

```diff
@@ -0,0 +1,85 @@
+# Project Environment
+
+## Introduction
+
+In here we'll discuss about the project environment. We'll discuss about the project architecture & folder structure, project environments (`Production`, `Development`, `Staging`), how secrets `(environment variables)` are handled, how to use environment variables for different environments,
+
+### Project Architecture & Folder Structure
+
+The project architecture is based on the [Next.js](https://nextjs.org/) framework. The project folder structure is as follows.
+
+```bash
+├── .github
+│   └── workflows
+│       └── ci.yml
+├── .next
+│   ├── cache
+│   ├── dist
+│   ├── server
+│   └── static
+├── .vscode
+│   └── settings.json
+├── components
+│   ├── common
+│   │   ├── Button
+│   │   │   ├── index.js
+│   │   │   └── styles.js
+│   │   ├── Card
+│   │   │   ├── index.js
+│   │   │   └── styles.js
+│   │   ├── Footer
+│   │   │   ├── index.js
+│   │   │   └── styles.js
+│   │   ├── Header
+│   │   │   ├── index.js
+│   │   │   └── styles.js
+│   │   ├── Layout
+│   │   │   ├── index.js
+│   │   │   └── styles.js
+│   │   └── Navbar
+│   │       ├── index.js
+│   │       └── styles.js
+│   └── pages
+│       ├── blog
+│       │   ├── [slug].js
+│       │   └── index.js
+│       ├── index.js
+│       └── post
+│           └── [...all].js
+├── next.config.js
+├── package.json
+├── pages
+│   ├── _app.js
+│   ├── _document.js
+│   ├── _error.js
+│   ├── blog
+│   │   ├── [slug].js
+│   │   └── index.js
+│   ├── index.js
+│   └── post
+│       └── [...all].js
+├── public
+│   ├── favicon.ico
+│   └── images
+│       └── logo.png
+├── README.md
+├── styles
+│   ├── global.js
+│   └── theme.js
+└── yarn.lock
```

Comment:
Why use YARN? Mixing YARN and NPM creates many problems

Reply:
yarn.lock file removed. Pls note that, this file structrue will change when we starting development. We have to update the docs after development started.